relay_messages_test.go

```diff
@@ -524,3 +550,41 @@ func TestLazyErrorCodes(t *testing.T) {
 		assert.Equal(t, ec, f.Code(), "Mismatch between error code and lazy frame's Code() method.")
 	})
 }
+
+// TODO(cinchurge): replace with e.g. decodeThriftHeader once we've resolved the import cycle
+func uint16KeyValToMap(tb testing.TB, buffer []byte) map[string]string {
+	rbuf := typed.NewReadBuffer(buffer)
+	nh := int(rbuf.ReadUint16())
+	retMap := make(map[string]string, nh)
+	for i := 0; i < nh; i++ {
+		keyLen := int(rbuf.ReadUint16())
+		key := rbuf.ReadBytes(keyLen)
+		valLen := int(rbuf.ReadUint16())
+		val := rbuf.ReadBytes(valLen)
+		retMap[string(key)] = string(val)
+	}
+	require.NoError(tb, rbuf.Err())
+	return retMap
+}
+
+func TestLazyCallReqContents(t *testing.T) {
+	cr := reqHasAll.reqWithParams(t, testCallReqParams{
+		arg2Buf: thriftarg2test.BuildKVBuffer(map[string]string{
+			"foo": "bar",
+			"baz": "qux",
+		}),
+		arg3Buf: []byte("some arg3 data"),
+	})
+
+	t.Run("checksum", func(t *testing.T) {
+		assert.Equal(t, ChecksumType(0x3), cr.checksumType)
+		assert.Equal(t, uint16(0x80), cr.checksumTypeOffset)
+	})
+
+	t.Run(".arg2()", func(t *testing.T) {
+		assert.Equal(t, map[string]string{
```

Comment:
nit: create a variable called headers with this map, and use it when we create / here?